test/unit/Cardano/Wallet/CoinSelection/LargestFirstSpec.hs

```diff
@@ -0,0 +1,294 @@
+import Test.Hspec
+    ( Expectation, Spec, describe, it, shouldBe, shouldSatisfy )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , Gen
+    , Property
+    , choose
+    , generate
+    , oneof
+    , property
+    , scale
+    , suchThat
+    , vectorOf
+    )
+import Test.QuickCheck.Monadic
+    ( monadicIO )
+
+import qualified Data.ByteString as BS
+import qualified Data.List as L
+import qualified Data.List.NonEmpty as NE
+import qualified Data.Map.Strict as Map
+
+spec :: Spec
+spec = do
+    describe "Coin selection : LargestFirst algorithm unit tests" $ do
+        it "happy path result in successful selection - a" $ do
+            (coinSelectionUnitTest
+                100
+                [10,10,17]
+                (17 :| [])
+                (Right [17])
+                )
+        it "happy path result in successful selection - b" $ do
+            (coinSelectionUnitTest
+                100
+                [12,10,17]
+                (1 :| [])
+                (Right [10])
+                )
+        it "NotEnoughMoney error expected when not enough coins" $ do
+            (coinSelectionUnitTest
+                100
+                [12,10,17]
+                (40 :| [])
+                (Left $ NotEnoughMoney 39 40)
+                )
+        it "NotEnoughMoney error expected when not enough coins and utxo not fragmented enough" $ do
+            (coinSelectionUnitTest
+                100
+                [12,10,17]
+                (40 :| [1,1,1])
+                (Left $ NotEnoughMoney 39 43)
+                )
+        it "UtxoNotEnoughFragmented error expected when enough coins and utxo not fragmented enough" $ do
+            (coinSelectionUnitTest
+                100
+                [12,20,17]
+                (40 :| [1,1,1])
+                (Left $ UtxoNotEnoughFragmented 3 4)
+                )
+        it "happy path with too strict maximumNumberOfInputs result in error" $ do
+            (coinSelectionUnitTest
+                1
+                [10,10,17]
+                (17 :| [1])
+                (Left $ MaximumInputsReached 1)
+                )
+    describe "Coin selection properties : LargestFirst algorithm" $ do
+        it "forall (UTxO, NonEmpty TxOut), running algorithm twice yields exactly the same result"
+            (property propDeterministic)
+        it "forall (UTxO, NonEmpty TxOut), there's at least as many selected inputs as there are requested outputs"
+            (property propAtLeast)
+
+{-------------------------------------------------------------------------------
+                 Properties and unit test generic scenario
+-------------------------------------------------------------------------------}
+
+coinSelectionUnitTest
+    :: Word64
+    -- ^ maximumNumberOfInputs
+    -> [Word64]
+    -- ^ utxo coins
+    -> NonEmpty Word64
+    -- ^ transaction outputs
+    -> Either CoinSelectionError [Word64]
+    -- ^ expecting CoinSelectionError or coins selected
```

Comment:
Nice. 
What about passing all the arguments above in a record type (instead of having comments here), so that at the call-site, it becomes even more readable ?

Reply:
done